### ISSUE TITLE
Better way to indicate lines that triggered exception

### ISSUE THREAD

**Issue by JoshVarty** [CodeConnect/AliveFeedback#106]
I received this message from a user today:

> Would it be possible to indicate which line threw and exception? In case of a test with multiple Asserts or simply just a method with more than one file / network operations, it’d be very useful to see which line failed without having to hover over the red dot and parse the exception info there.

Perhaps we could make it so when you clicked on the dot it took you to the line or something? Might be a quick way to jump around easily.

**Comment by JoanComasFdz** [comment]
If you use NCrunch you'll see he is able to do that, so it is possible to find out and mark the line. It would be a great feature :)

**Comment by vbandi** [comment]
"Perhaps we could make it so when you clicked on the dot it took you to the line or something? Might be a quick way to jump around easily." - sorry, I'd rather not click anything. While it'd certainly help, nothing beats just looking at the proper line highlighted.
(BTW, I was the "user" suggesting this, thanks for following up!)

**Comment by AmadeusW** [comment]
Sometimes the exception happens outside of the method that you're looking at, and that's when users may want to jump there.

I agree with you that a simple UI beats all kinds of clicking. You might be interested in a new feature that we are getting ready to release soon: it's a line highlighter that shows which lines were executed.

It shows which `if` branches or `foreach` loops were taken, and you can use it to spot where the exception happened - in the first line that's shaded (marked as not executed).

![2016-03-21_14h51_38](https://cloud.githubusercontent.com/assets/1673956/13935470/aeaf34cc-ef74-11e5-9f4c-45ad43d28ffc.png)

In the future we will see if we can take advantage of PDBs that would allow us to map exceptions to line numbers and specific members. However, generating PDBs is slow and we want to focus on achieving a very fast refresh speed.

**Comment by vbandi** [comment]
Yeah, this could help with Unit Tests, which are mostly linear. Admittedly knowing nothing about PDB files though: can't you extract the required information from the exception stack trace?

**Comment by AmadeusW** [comment]
Without PDB we get a stack trace without line numbers - just with the method names. 
This means that we can jump to the method that threw the exception, or at least get as close as possible if it exists in a referenced library, but we aren't able to pin-point the specific line.

**Comment by vbandi** [comment]
I see. Well, in this case I am looking forward to the highlighter feature.

**Comment by lundmikkel** [comment]
I would just like to add my two cents.

Currently, it is a bit annoying when a long method throws an exception and you have no idea where it went wrong, cause you just get a red dot. Then you have to actually run the unit test to see where it happened. This kind of defeats the idea of Alive.

I do expect that sometimes the exception is thrown in the test method – not the method I'm trying to continue at – but there is no indication that this is the case.

**Comment by JoshVarty** [comment]
I agree and this is something that we're actively working toward supporting. It's going to be a bit of a challenge to support properly, but it's one of our biggest priorities now that we're wrapping up out of process execution.
